analyzer/codechecker_analyzer/analysis_manager.py

```diff
@@ -444,6 +444,27 @@ def __cleanup_timeout():
     return __cleanup_timeout
 
 
+def collect_ctu_involved_files(result_handler, source_analyzer, output_dir):
+    """
+    This function collects the list of source files involved by CTU analysis.
+    The list of files are written to output_dir.
+    """
+    if source_analyzer.ANALYZER_NAME != ClangSA.ANALYZER_NAME:
+        return
+
+    involved_files = set()
+
+    involved_files.update(source_analyzer.get_analyzer_mentioned_files(
+        result_handler.analyzer_stdout))
+    involved_files.update(source_analyzer.get_analyzer_mentioned_files(
+        result_handler.analyzer_stderr))
+
+    if involved_files:
```

Comment:
It is possible that previously this set wasn't empty, so we created an involved file, but the next time it was empty, so we do nothing. Do not we need to remove the previous file? So something similar to this:
```py
out = os.path.join(output_dir, result_handler.analyzer_action_str)
if involved_files:
    ...
else if os.path.exists(out):
    os.remove(out)
```

Reply:
I don't understand this comment. This directory contains generated files for for those TUs which involve some other source files during CTU analysis. It doesn't matter if there were files here with the same name because those will be rewritten. The content of this directory behaves the same way as `failed` directory.

Reply:
@bruntib Okay, I try to explain my problem again. Lets assume that you analyzed the same TU two times. The first time the `involved_files` variable contains some file (`lib.cpp`) so you will create a file (`result_handler.analyzer_action_str`) which will contain the involved files (`lib.cpp`). If you change something in your code, you analyze your TU again and the `involved_files` set is empty you will do nothing. But in the `output_dir` there will be a file for this TU (`result_handler.analyzer_action_str`) which will contain the `lib.cpp` involved file from the previous analysis.
My question was that in this case don't we need to remove this file if it's exist and the `involved_files` set is empty?

Reply:
Well, it doesn't cause any problem if those files are not removed. This is in analogy with failed ZIPs and .plist files: those are also not removed when you have lass analyzed TUs. Though we can remove these and the failed ZIPs if we want to gain some free space, though it's not that significant I think. But I'll check it and do this removal for failed ZIPs accordingly in a next commit.

Reply:
Ok, I did it.